include/NovelRT.Interop/Ecs/NrtComponentBufferMemoryContainer.h

```diff
@@ -0,0 +1,67 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#ifndef NOVELRT_NRTCOMPONENTBUFFERMEMORYCONTAINER_H
+#define NOVELRT_NRTCOMPONENTBUFFERMEMORYCONTAINER_H
+
+#include "../NrtInteropUtils.h"
+#include "NrtEcsUtils.h"
+#include "NrtSparseSetMemoryContainer.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    typedef struct ComponentBufferMemoryContainerHandle* NrtComponentBufferMemoryContainer;
+
+    typedef struct ComponentBufferImmutableDataViewHandle* NrtComponentBufferMemoryContainer_ImmutableDataView;
+
+    typedef (*NrtComponentUpdateFnPtr)(NrtSparseSetMemoryContainer_ByteIteratorView,
+                                       NrtSparseSetMemoryContainer_ByteIteratorView,
+                                       size_t);
+
+    NrtComponentBufferMemoryContainer Nrt_ComponentBufferMemoryContainer_Create(size_t poolSize,
+                                                                                NrtComponentUpdateFnPtr fnPtr);
+
+    void Nrt_ComponentBufferMemoryContainer_PrepContainerForFrame(NrtComponentBufferMemoryContainer container,
+                                                                  NrtEntityIdVector entitiesToDelete);
+
+    NrtComponentBufferMemoryContainer_ImmutableDataView Nrt_ComponentBufferMemoryContainer_GetDeleteInstructionState(
+        NrtComponentBufferMemoryContainer container);
+
+    NrtResult Nrt_ComponentBufferMemoryContainer_PushComponentUpdateInstruction(
+        NrtComponentBufferMemoryContainer container,
+        size_t poolId,
+        NrtEntityId entity,
+        const void* componentData);
+
+    NrtResult Nrt_ComponentBufferMemoryContainer_GetComponent(
+        NrtComponentBufferMemoryContainer container,
+        NrtEntityId entity,
+        NrtComponentBufferMemoryContainer_ImmutableDataView* outputResult);
+
+    NrtComponentBufferMemoryContainer_ImmutableDataView Nrt_ComponentBufferMemoryContainer_GetComponentUnsafe(
+        NrtComponentBufferMemoryContainer container,
+        NrtEntityId entity);
+
+    NrtBool Nrt_ComponentBufferMemoryContainer_HasComponent(NrtComponentBufferMemoryContainer container,
+                                                            NrtEntityId entity);
+
+    size_t Nrt_ComponentBufferMemoryContainer_GetImmutableDataLength(NrtComponentBufferMemoryContainer container);
+
+    // clang-format off
```

Comment:
Is there a reason for turning formatting off here?

Reply:
I've been doing it in any place where we have begin() and end() functions, but since its a clang-tidy thing now, not really sure if this does what I intended.

Reply:
Agreed, you may be same to remove this as the definition itself shouldn't be affected by clang-format.

If It breaks afterwards we can always put it back